src/Segments/VCS.hs

```diff
@@ -87,6 +90,13 @@ gitCommitsAheadCount = runMaybeT $ do
   count  <- MaybeT $ readProcess "git" ["rev-list", "--count", remote ++ "/" ++ branch ++ ".." ++ branch]
   MaybeT . return . showNZ . read $ count
 
+gitCommitsBehindCount :: IO (Maybe String)
+gitCommitsBehindCount = runMaybeT $ do
+  branch <- MaybeT gitBranch
+  remote <- MaybeT $ readProcess "git" ["config", "--get", "branch." ++ branch ++ ".remote"]
+  count  <- MaybeT $ readProcess "git" ["rev-list", "--count", branch ++ ".." ++ remote ++ "/" ++ branch]
```

Comment:
`gitCommitsBehindCount` returns Nothing if the branch does not have a remote. 

I haven't dived deep into the code but after a first quick analysis, I think `branch` will be an empty string, and in the next stage it will fail when trying to calculate the count, therefore returning Nothing.

I can do a more thorough analysis later today if you think it's necessary.

Reply:
Ah, so it relies on `readProcess` returning `IO Nothing`. Could be done more efficiently, but this is fine.

Reply:
Ok. Would you mind explaining what you mean in more detail?

I'll have more PRs ready soon and I could also improve this